frontend/frontend.php

```diff
@@ -27,10 +48,35 @@ public function __construct() {
 			return;
 		}
 
+		add_action( 'send_headers', [ $this, 'prepare_script' ], 10 );
 		add_action( 'wp_head', [ $this, 'script' ], 90 );
 		add_action( 'comment_post', [ $this, 'track_comment' ], 10, 2 );
 	}
 
+	/**
+	 * Prepares our script tags. Needed to figure out if we need to send out CSP headers.
+	 */
+	public function prepare_script() {
+		if ( is_preview() ) {
+			return;
+		}
+
+		$this->inline_script  = $this->goal_tracking();
+		$this->inline_script .= $this->outbound_tracking();
+		$this->inline_script .= $this->disable_cookies();
+
+		$this->extra_nonce = wp_create_nonce( 'clicky_extra_script_nonce' );
+		$this->names_nonce = wp_create_nonce( 'clicky_names_script_nonce' );
+
+		/**
+		 * Action: 'yoast/clicky/csp' - Allows CSP developers to easily get the nonces for the inline scripts Clicky outputs.
+		 *
+		 * @param string $extra_nonce The nonce for the script tag Clicky might output.
```

Comment:
```suggestion
		 * @param string $extra_nonce The nonce for the script tag Clicky might output.
```
could we rename `$extra_nonce` to something more descriptive e.g. `$inline_script_nonce` ?
that would apply to all occurrences of `extra`🕵️